### DISCUSSION TITLE
Regression Not Working

### DISCUSSION THREAD

**Discussion by tahsin314** [Lightning-AI/pytorch-lightning#6949] [code help: CV]
I'm working on a regression task. My model is supposed to take an image of retina as input and predict the severity level of an abnormality. The severity level can be any integer between 0 and 4. [Here](https://github.com/tahsin314/Diabetic_Retinopathy-Detection-Pattern-Recognition-Course-Project/blob/torch_lightning/train.py#L156) is my lightning module. The code works perfectly if I use `ordinal regression` but does not converge when I use `regression` and `MSELoss`. ([Train Loss](https://wandb.ai/tahsin/Diabetic_Retinopathy/reports/Shared-panel-21-04-10-16-04-18--Vmlldzo1OTgxODk) and [Validation Loss](https://wandb.ai/tahsin/Diabetic_Retinopathy/reports/Shared-panel-21-04-10-16-04-38--Vmlldzo1OTgxOTI)). I've tried varying learning rate from `10^-5` to `10^1` but there isn't any improvement. I've also tried to overfit my model but it didn't help either. It would be a great help if someone can look into it. Thanks.
